syntax = "proto3";

package admin;

enum Result {
  SUCCESS = 0;
  USER_NOT_FOUND = 1;
  MAILBOX_NOT_FOUND = 2;
}

message AppendRequest {
  string user = 1;
  string mailbox = 2;
  bytes data = 3;
  repeated string flags = 4;
  uint64 when = 5;
}

message AppendResponse {
  Result result = 1;
  uint32 validity = 2;
  uint32 uid = 3;
}

service Admin {
  rpc Append (AppendRequest) returns (AppendResponse) {}
}
